platform/platform.go

```diff
@@ -116,3 +117,47 @@ func transfer(in io.Reader, m Machine, localPath string) error {
 
 	return nil
 }
+
+func NewMachines(c Cluster, userdatas []string) ([]Machine, error) {
+	var wg sync.WaitGroup
+	var machs []Machine
+
+	n := len(userdatas)
+
+	if n <= 0 {
+		return nil, fmt.Errorf("must provide one or more userdatas")
+	}
+
+	mchan := make(chan Machine, n)
+	errchan := make(chan error, n)
+
+	for i := 0; i < n; i++ {
+		ud := userdatas[i]
+		wg.Add(1)
+		go func() {
+			m, err := c.NewMachine(ud)
+			if err != nil {
+				errchan <- err
+			}
+			mchan <- m
+			wg.Done()
+		}()
+	}
+
+	wg.Wait()
+	close(mchan)
+	close(errchan)
+
+	for m := range mchan {
+		machs = append(machs, m)
+	}
+
+	if firsterr, ok := <-errchan; ok {
+		for _, m := range machs {
+			m.Destroy()
```

Comment:
If you intended to potentially get nil machines, you should probably be testing if m is nil before calling Destroy() here.